neotools/cli.py

```diff
@@ -9,8 +9,26 @@
 
 logger = logging.getLogger(__name__)
 
+class BasedIntParamType(click.ParamType):
+    name = "integer"
+
+    def convert(self, value, param, ctx):
+        if isinstance(value, int):
+            return value
+
+        try:
+            if value[:2].lower() == "0x":
+                return int(value[2:], 16)
+            elif value[:1] == "0":
+                return int(value, 8)
```

Comment:
I think that the octal numbers are rarely used for anything else other than permissions. Also, they may be a source of error if a user enters a zero-padded integer.
Do you mind removing them or making the prefix unambiguous? For example, Rust uses prefixes 0x, 0o, 0b for the hex, octal and binary numbers.

Reply:
This code is from the `click` docs. I almost removed them but then forgot.